packages/hoppscotch-sh-admin/src/helpers/userManagement.ts

```diff
@@ -0,0 +1,121 @@
+import { useToast } from '~/composables/toast';
+import { getI18n } from '~/modules/i18n';
+import { UserDeletionResult } from './backend/graphql';
+import { ADMIN_CANNOT_BE_DELETED, USER_IS_OWNER } from './errors';
+
+type IndividualActionInput = {
+  type: 'individual';
+  metadata: null;
+};
+type BulkActionInput = {
+  type: 'bulk';
+  metadata: {
+    areMultipleUsersSelected: boolean;
+    deletedIDs: string[];
+  };
+};
+
+type IndividualActionResult = {
+  data: null;
+};
+type BulkActionResult = {
+  data: { timeoutID: NodeJS.Timeout | null };
+};
+
+type HandleUserDeletion = {
+  (
+    deletedUsersList: UserDeletionResult[],
+    action: IndividualActionInput | BulkActionInput
+  ): IndividualActionResult | BulkActionResult;
+};
+
+const t = getI18n();
+const toast = useToast();
+
+export const handleUserDeletion: HandleUserDeletion = (
```

Comment:
why do we need to have seperate `bulk` v/s `individual` seperation. we could define a function which takes, `deletedUsersList` and derive the other things from it. 

```typescript
const handleUserDeletionCombined = (
  // isn't this the only thing that is required ? can't all the other things be derived from this?
  deletedUsersList: UserDeletionResult[]
)
```

```typescript
// can't this be used instead of a bulk property?
const hasMultipleUsers = deletedUsersList.length > 1;
```